description: clang 7 has trouble deducing auto type in lambda expressions
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/bindings/core/v8/serialization/serialized_script_value.h
+++ b/third_party/blink/renderer/bindings/core/v8/serialization/serialized_script_value.h
@@ -268,12 +268,16 @@ class CORE_EXPORT SerializedScriptValue
   MessagePortChannelArray& GetStreamChannels() { return stream_channels_; }
 
   bool IsLockedToAgentCluster() const {
+    bool is_locked = false;
+    for (const auto& entry : attachments_) {
+      if (entry.value->IsLockedToAgentCluster()) {
+        is_locked = true;
+      }
+    }
+
     return !wasm_modules_.IsEmpty() ||
            !shared_array_buffers_contents_.IsEmpty() ||
-           std::any_of(attachments_.begin(), attachments_.end(),
-                       [](const auto& entry) {
-                         return entry.value->IsLockedToAgentCluster();
-                       });
+           is_locked;
   }
 
   // Returns true after serializing script values that remote origins cannot
--- a/third_party/blink/renderer/core/layout/ng/ng_physical_container_fragment.cc
+++ b/third_party/blink/renderer/core/layout/ng/ng_physical_container_fragment.cc
@@ -122,11 +122,9 @@ void NGPhysicalContainerFragment::AddOut
       // Don't add |Children()|. If |this| has |NGFragmentItems|, children are
       // either line box, which we already handled in items, or OOF, which we
       // should ignore.
-      DCHECK(std::all_of(PostLayoutChildren().begin(),
-                         PostLayoutChildren().end(), [](const NGLink& child) {
-                           return child->IsLineBox() ||
-                                  child->IsOutOfFlowPositioned();
-                         }));
+      for (const NGLink& child : PostLayoutChildren()) {
+        DCHECK(child->IsLineBox() || child->IsOutOfFlowPositioned());
+      }
       return;
     }
   }
--- a/third_party/blink/renderer/modules/accessibility/ax_object.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_object.cc
@@ -1815,11 +1815,12 @@ bool AXObject::IsSubWidget() const {
 
       // Otherwise it's only a subwidget if it's in a grid or treegrid,
       // not in a table.
-      return std::any_of(
-          AncestorsBegin(), AncestorsEnd(), [](const AXObject& ancestor) {
-            return ancestor.RoleValue() == ax::mojom::Role::kGrid ||
-                   ancestor.RoleValue() == ax::mojom::Role::kTreeGrid;
-          });
+      for (auto ancestor = AncestorsBegin(); ancestor != AncestorsEnd(); ++ancestor) {
+        if (ancestor->RoleValue() == ax::mojom::Role::kGrid ||
+            ancestor->RoleValue() == ax::mojom::Role::kTreeGrid)
+          return true;
+      }
+      return false;
 
     case ax::mojom::Role::kListBoxOption:
     case ax::mojom::Role::kMenuListOption:
@@ -4043,11 +4044,11 @@ bool AXObject::SupportsARIAReadOnly() co
 
   if (ui::IsCellOrTableHeader(RoleValue())) {
     // For cells and row/column headers, readonly is supported within a grid.
-    return std::any_of(
-        AncestorsBegin(), AncestorsEnd(), [](const AXObject& ancestor) {
-          return ancestor.RoleValue() == ax::mojom::Role::kGrid ||
-                 ancestor.RoleValue() == ax::mojom::Role::kTreeGrid;
-        });
+    for (auto ancestor = AncestorsBegin(); ancestor != AncestorsEnd(); ++ancestor) {
+        if (ancestor->RoleValue() == ax::mojom::Role::kGrid ||
+            ancestor->RoleValue() == ax::mojom::Role::kTreeGrid)
+          return true;
+    }
   }
 
   return false;
